cpu/cc2538/lpm.c

```diff
@@ -284,6 +321,25 @@ lpm_enter()
     return;
   } else if(duration >= DEEP_SLEEP_PM2_THRESHOLD && max_pm == 2) {
     /* Long sleep duration and PM2 is allowed. Use it */
+
+    /* CC2538 silicon errata:
+       Possible Incorrect Value of Clock Dividers after PM2 and PM3 */
+#if SYS_CTRL_SYS_DIV == SYS_CTRL_CLOCK_CTRL_SYS_DIV_32MHZ && \
+    SYS_CTRL_IO_DIV == SYS_CTRL_CLOCK_CTRL_IO_DIV_32MHZ
+    /* Set both sys + io clock divisor to 16MHz */
+    REG(SYS_CTRL_CLOCK_CTRL) |= SYS_CTRL_CLOCK_CTRL_SYS_DIV_16MHZ
+      + SYS_CTRL_CLOCK_CTRL_IO_DIV_16MHZ;
+    reset_div = 1;
+#elif SYS_CTRL_SYS_DIV == SYS_CTRL_CLOCK_CTRL_SYS_DIV_32MHZ
+    /* Set sys clock divisor to 16MHz */
+    REG(SYS_CTRL_CLOCK_CTRL) |= SYS_CTRL_CLOCK_CTRL_SYS_DIV_16MHZ;
+    reset_div = 1;
+#elif SYS_CTRL_IO_DIV == SYS_CTRL_CLOCK_CTRL_IO_DIV_32MHZ
+    /* Set io clock divisor to 16MHz */
+    REG(SYS_CTRL_CLOCK_CTRL) |= SYS_CTRL_CLOCK_CTRL_IO_DIV_16MHZ;
+    reset_div = 1;
+#endif
+
```

Comment:
Simplify and move to `select_16_mhz_rcosc()`. See my previous comments for more details.